### PR TITLE
Create check for html entities

### PR DESCRIPTION
**Pull request by nickbrowne**
This copies the code from the html entity rake task to a Check class, and incorporates that Check into the configurable RakeTask.

### CODE REVIEW

File: lib/i18n/hygiene/wrapper.rb [hunk 1]
```diff
@@ -17,7 +18,7 @@ def keys_to_check(locale = :en)
       end
 
       def locales
-        translations.keys
+        translations.keys.select { |key| @locales.include?(key) }
```
Comment (nickbrowne) [inline lib/i18n/hygiene/wrapper.rb]: I find this a bit odd, but it works as you'd expect, we can refactor it later.

**Comment by nickbrowne** [comment]
[[i18n-hygiene] Create UnexpectedHtmlEntity Check class (check_entities rake task)](https://trello.com/c/6mtnSNMe/7771-i18n-hygiene-create-unexpectedhtmlentity-check-class-checkentities-rake-task)

File: lib/i18n/hygiene/wrapper.rb [hunk 2]
```diff
@@ -8,7 +8,8 @@ module Hygiene
     # queryable.
     class Wrapper
 
-      def initialize(keys_to_skip:)
+      def initialize(keys_to_skip: [], locales: ::I18n.available_locales)
```
Comment (yob) [inline lib/i18n/hygiene/wrapper.rb:11]: Can we trust `::I18n.available_locales` to always return something sensible?

Is it always set? I think we only started setting it in our apps recently, and before that it *may* have been an empty array?

Comment (nickbrowne) [reply lib/i18n/hygiene/wrapper.rb:11]: It seems to be set automatically when you load/store translations, for example in the spec I wrote:
```ruby
[:fr, :es].each do |locale|
  ::I18n.backend.send(:store_translations, locale, abuse_report: { button: { submit: "x" }})
end
```

If you omit the `locales: [:en, :fr]` in the constructor, it includes them all via `I18n.available_locales`.

I might expand that test to make that more obvious.

File: lib/i18n/hygiene/wrapper.rb (hunk 1 above)
Comment (nickbrowne) [reply lib/i18n/hygiene/wrapper.rb]: Actually this can just be `@locales`

File: lib/i18n/hygiene/checks/html_entity.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+require 'i18n/hygiene/wrapper'
+
+module I18n
+  module Hygiene
+    module Checks
+      class HtmlEntity < Base
+        def run
+          puts "Checking for phrases that contain entities but probably shouldn't..."
+
+          wrapper = I18n::Hygiene::Wrapper.new(locales: config.locales)
```
Comment (yob) [inline lib/i18n/hygiene/checks/html_entity.rb]: if `config.locales` contains `[:fr,:es]`, does that mean this task won't check the main locale (`:en` for us) as well?

Comment (nickbrowne) [reply lib/i18n/hygiene/checks/html_entity.rb]: Indeed, I'll expand the integration tests to catch this situation.

**Comment by nickbrowne** [comment]
Rebased to pull in a few features needed to allow the entity check to also scan the primary locale.

**Comment by yob** [comment]
I've tested this in donations with the following config for the new rake task and confirmed it fails when there's entities in the `en` and `fr` locales:

```ruby
I18n::Hygiene::RakeTask.new(:hygiene) do |config|
  config.keys_to_skip = [
    "helpers.submit.create",
    "helpers.select.prompt", 
    "helpers.submit.submit",
    "helpers.submit.update",
   ]
  config.directories = ["app", "lib"]
  config.locales = [:fr]
  config.primary_locale = :en
end

```

**Pull request merged**